pkg/wal/wal.go

```diff
@@ -0,0 +1,46 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Package wal (Write-ahead logging) is an independent component to ensure data reliability.
+package wal
+
+// Log stands for a write-ahead logging instance.
+type Log struct{}
+
+// Options for Write-ahead Logging.
+type Options struct{}
+
+// Segment stands for a segment instance of Write-ahead log.
+type Segment struct{}
+
+// Wal include exposed interfaces.
+type Wal interface {
+	// New creates a Log instance in the specified root directory.
+	New(path string, opts *Options) (*Log, error)
```

Comment:
Could you transfer it to a package-scoped function? This way is a go conventional path to build a constructor.